Shokofin/Utils/GuardedMemoryCache.cs

```diff
@@ -18,7 +17,9 @@ sealed class GuardedMemoryCache : IDisposable, IMemoryCache
 
     private IMemoryCache Cache;
 
-    private readonly ConcurrentDictionary<object, SemaphoreSlim> Semaphores = new();
+    private static AsyncKeyedLockOptions AsyncKeyedLockOptions = new() { PoolSize = 20, PoolInitialFill = 1 };
```

Comment:
Also, reading up on what pooling is, I would increase this from 20 to _at least_ 100. If we take an average series of 13 episodes, then add the same amount of files, then add in the series and season, then we get an average size of _up to_ 28 locks needed per series. Now if we assume the default concurrency of 4 that give us potentially _up to_ 112 locks being in used at the same time, give or take. Do you agree or disagree on this?

Reply:
I found 20 as the sweet spot even in situations of much higher load. The higher you increase that the more items you risk staying in memory (in the pool) indefinitely. However, realistically, I doubt you would reach that locks because you will get bottlenecked by your Environment.ProcessorCount much earlier than you'd hit that. If you want to increase it though, up to you.

Reply:
I'll trust you on that and leave it at 20 for now. Thanks for the clarification/information.